### PR TITLE
adding tests to annotationKit

### PR DESCRIPTION
**Pull request by aradiquez**
i try to test as much as i could, also change the header on some files just to keep the standard

//cc @Lucashuang0802 @marinaserranomontes @marko07

### CODE REVIEW

**Comment by Lucashuang0802** [comment]
Close for submitting another clean PR.

File: ios/AnnotationAccPackKit/OTAnnotationAccPackKit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -281,6 +299,63 @@
...
+			sourceTree = "<group>";
+		};
+		3FE8672E1D416449006D5218 /* Toolbar */ = {
+			isa = PBXGroup;
+			children = (
+				3FE8672F1D41667F006D5218 /* OTAnnotationToolbarButtonTests.m */,
+			);
+			name = Toolbar;
+			sourceTree = "<group>";
+		};
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationAccPackKit.xcodeproj/project.pbxproj]: What's this change above?

Comment (aradiquez) [inline ios/AnnotationAccPackKit/OTAnnotationAccPackKit.xcodeproj/project.pbxproj]: i did a folder structure similar to the kit just to be a little bit more clear in the testing distribution.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationKitTests.m [hunk 2]
```diff
@@ -1,39 +1,29 @@
...
+#import <Kiwi/Kiwi.h>
+#import "OTAnnotationKitBundle.h"
+
+SPEC_BEGIN(OTAnnotationKitTests)
+
+describe(@"Initialization of OTAnnotationKitBundle", ^(){
+    
+    context(@"An instance of OTAnnotationKitBundle", ^(){
+        
+        OTAnnotationKitBundle *annotationBundle = [[OTAnnotationKitBundle alloc] init];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationKitTests.m]: You don't need to initialize it.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationNative/OTAnnotationTextView.m [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    
+    context(@"A new instance of OTAnnotationTextView", ^{
+        
+        it(@"gesture variables should be FALSE before init", ^{
+            
+            OTAnnotationTextView *annotationsText = [OTAnnotationTextView alloc];
+            
+            [[theValue(annotationsText.isResizable) should] equal:@0];
+            [[theValue(annotationsText.isDraggable) should] equal:@0];
+            [[theValue(annotationsText.isRotatable) should] equal:@0];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationNative/OTAnnotationTextView.m]: You should check boolean value.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationNative/OTAnnotationTextView.m [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        });
+        
+        it(@"should set the gesture variables after initialize", ^{
+            
+            OTAnnotationTextView *annotationsText = [[OTAnnotationTextView alloc] initWithText:@"" textColor:[UIColor greenColor] fontSize:0.0f];
+            
+            [[theValue(annotationsText.isResizable) should] equal:@1];
+            [[theValue(annotationsText.isDraggable) should] equal:@1];
+            [[theValue(annotationsText.isRotatable) should] equal:@1];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationNative/OTAnnotationTextView.m]: You should check boolean value.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationViewTests.m [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            [[annotationView.annotationDataManager shouldNot] beNil];
+        });
+    });
+});
+
+describe(@"Methods of OTAnnotationView", ^{
+    
+    let(annotationView, ^{
+        return [[OTAnnotationView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
+    });
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationViewTests.m]: What's this again? 😄

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationViewTests.m [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    
+    context(@"Method", ^{
+    
+        it(@"commitCurrentAnnotatable should set currentAnnotatable to be nil", ^{
+            [annotationView commitCurrentAnnotatable];
+            [[(id)annotationView.currentAnnotatable should] beNil];
+        });
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationViewTests.m]: This test is not correct, can you read the docs?

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationViewTests.m [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        
+        it(@"captureScreen result image should be same bounds as mainScreen", ^{
+            CGRect screenRectSize = [[UIScreen mainScreen] bounds];
+            UIImage *image = [annotationView captureScreen];
+            [[theValue(image.size.width) should] equal:theValue(screenRectSize.size.width)];
+            [[theValue(image.size.height) should] equal:theValue(screenRectSize.size.height)];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationViewTests.m]: Just check whether it's `nil` or not, no need to check the size.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationToolbarButtonTests.m [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        
+        OTAnnotationToolbarView *annotationView = [[OTAnnotationToolbarView alloc]  init];
+        
+        it(@"should not be nil", ^{
+            [[annotationView shouldNot] beNil];
+        });
+    });
+});
+
+SPEC_END
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationToolbarButtonTests.m]: Let's go ahead to remove this file. It's not used. Thanks. 😄

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationTextView.m [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    context(@"A new instance of OTAnnotationTextView", ^{
+        
+        it(@"gesture variables should be FALSE before init", ^{
+            
+            OTAnnotationTextView *annotationsText = [OTAnnotationTextView alloc];
+            
+            [[theValue(annotationsText.isResizable) should] equal:theValue(NO)];
+            [[theValue(annotationsText.isDraggable) should] equal:theValue(NO)];
+            [[theValue(annotationsText.isRotatable) should] equal:theValue(NO)];
+        });
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationTextView.m]: I don't think we provide `-init`, and you forget the `init`.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationTextView.m [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+SPEC_BEGIN(OTAnnotationTextViewTest)
+
+describe(@"Initialization of OTAnnotationTextView", ^(){
+    
+    context(@"An instance of OTAnnotationTextView", ^(){
+        
+        let(annotationText, ^{
+            return [OTAnnotationTextView alloc];
+        });
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationTextView.m]: And what's this again?

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m [hunk 11]
```diff
@@ -10,12 +10,55 @@
...
+context(@"OTAnnotationScrollView", ^(){
     
     describe(@"An instance of OTAnnotationScrollView", ^(){
+        OTAnnotationScrollView *annotationView = [OTAnnotationScrollView alloc];
         
-        OTAnnotationScrollView *annotationView = [[OTAnnotationScrollView alloc] init];
-        [[annotationView shouldNot] beNil];
+        it(@"should not be nil", ^{
+            [[[annotationView init] shouldNot] beNil];
+        });
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m]: You should not use `init` here. You should initialize it once you allocate

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScreenCaptureViewTests.m [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+SPEC_BEGIN(OTAnnotationScreenCaptureViewTests)
+
+context(@"Initialization of OTAnnotationScreenCaptureView", ^(){
+    
+    describe(@"An instance of OTAnnotationScreenCaptureView", ^(){
+        
+        OTAnnotationScreenCaptureModel *screenCaptureModel = [OTAnnotationScreenCaptureModel alloc];
+        
+        it(@"should not be nil", ^{
+            [[[screenCaptureModel init] shouldNot] beNil];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScreenCaptureViewTests.m]: `alloc` and `init` should be used as a pair.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScreenCaptureViewControllerTests.m [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+context(@"Initialization of OTAnnotationScreenCaptureViewController", ^(){
+    
+    describe(@"An instance of OTAnnotationScreenCaptureViewController", ^(){
+        
+        OTAnnotationScreenCaptureViewController *screenCaptureCtr = [OTAnnotationScreenCaptureViewController alloc];
+        UIImage *image = [[UIImage alloc] init];
+        
+        it(@"should not be nil", ^{
+            [[[screenCaptureCtr init] shouldNot] beNil];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScreenCaptureViewControllerTests.m]: You have two `init` here!

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationColorPickerViewTests.m [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+            [[annotationColorPickerView shouldNot] beNil];
+        });
+    });
+    
+    describe(@"OTAnnotationColorPickerView", ^{
+        
+        OTAnnotationColorPickerView *annotationColorPickerView = [[OTAnnotationColorPickerView alloc]  init];
+        
+        it(@"should be able to return background color", ^{
+            [[[annotationColorPickerView selectedColor] shouldNot] beNil];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationColorPickerViewTests.m]: This won't pass, will it? You haven't even give it a color.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationEditTextViewControllerTests.m [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+SPEC_BEGIN(OTAnnotationEditTextViewControllerTests)
+
+context(@"Initialization of OTAnnotationEditTextViewController", ^(){
+    
+    describe(@"An instance of OTAnnotationEditTextViewController", ^(){
+        
+        OTAnnotationEditTextViewController *editTextView = [OTAnnotationEditTextViewController alloc];
+        
+        it(@"should not be nil", ^{
+            [[[editTextView init] shouldNot] beNil];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationEditTextViewControllerTests.m]: Again, we should not separate `init` and `alloc`.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationEditTextViewControllerTests.m [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+        });
+        
+        it(@"init with text should not be nil", ^{
+            [[[editTextView initWithText:@"" textColor:[UIColor blackColor] fontSize:0.0f] shouldNot] beNil];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationEditTextViewControllerTests.m]: We should not separate `init` and `alloc`.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScreenCaptureViewTests.m [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+        });
+        
+        it(@"should be able to init with Image", ^{
+            UIImage *image = [[UIImage alloc] init];
+            OTAnnotationScreenCaptureModel *screenCaptureModel = [[OTAnnotationScreenCaptureModel alloc] initWithSharedImage:image sharedDate:[NSDate date]];
+            [[screenCaptureModel shouldNot] beNil];
+            [[screenCaptureModel.sharedImage should] equal:image];
+        });
+    });
+});
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScreenCaptureViewTests.m]: Can you combine `OTAnnotationScreenCaptureViewControllerTests` and this into one test set?

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m [hunk 18]
```diff
@@ -10,12 +10,54 @@
...
 
-context(@"Initialization of OTAnnotationScrollView", ^(){
+context(@"OTAnnotationScrollView", ^(){
     
     describe(@"An instance of OTAnnotationScrollView", ^(){
         
-        OTAnnotationScrollView *annotationView = [[OTAnnotationScrollView alloc] init];
-        [[annotationView shouldNot] beNil];
+        it(@"should not be nil", ^{
+            [[[[OTAnnotationScrollView alloc] init] shouldNot] beNil];
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m]: I mean, you should have a var to refer it. Keep this line `OTAnnotationScrollView *annotationView = [[OTAnnotationScrollView alloc] init];`

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m [hunk 19]
```diff
@@ -10,12 +10,54 @@
...
+        
+        it(@"should not be nil when init with frame", ^{
+            [[[[OTAnnotationScrollView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)] shouldNot] beNil];
+        });
+    });
+    
+    describe(@"An instance of OTAnnotationScrollView should set variables", ^{
+        let(annotationView, ^{
+            return [[OTAnnotationScrollView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
+        });
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m]: What's does this block do?

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m [hunk 20]
```diff
@@ -10,12 +10,54 @@
...
+        it(@"isZoomEnabled should be NO", ^{
+            [annotationView setZoomEnabled:NO];
+            [[theValue(annotationView.isZoomEnabled) should] equal:theValue(NO)];
+        });
+        
+        it(@"toolbar needs to be initialize first", ^{
+            [[annotationView.toolbarView should] beNil];
+        });
+        
+        it(@"toolbar needs to be initialize first", ^{
```
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m]: Can you change the description?

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationColorPickerViewTests.m (hunk 14 above)
Comment (aradiquez) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationColorPickerViewTests.m]: `selectedColor` is a method that returns a color

Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationColorPickerViewTests.m]: But you haven't selected anything. You just initialized it.

Comment (aradiquez) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationColorPickerViewTests.m]: yes, but init method set the background color that this function is returning

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m (hunk 19 above)
Comment (aradiquez) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationScrollViewTests.m]: about this block i found it reading examples and documents related to kiwi testing, and i thought it was like `rspec` that had this `let` syntax to eager load variables on request, but i think it's not the same here.

File: ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationColorPickerViewTests.m (hunk 14 above)
Comment (Lucashuang0802) [inline ios/AnnotationAccPackKit/OTAnnotationKitTests/OTAnnotationColorPickerViewTests.m]: indeed 👍

**Pull request merged**